src/Custom/RealtimeConversation/Internal/AsyncWebsocketMessageEnumerator.cs

```diff
@@ -26,8 +27,12 @@ public AsyncWebsocketMessageResultEnumerator(WebSocket webSocket, CancellationTo
 
     public ValueTask DisposeAsync()
     {
-        ArrayPool<byte>.Shared.Return(_receiveBuffer);
-        _webSocket?.Dispose();
+        if (!_disposed)
+        {
+            ArrayPool<byte>.Shared.Return(_receiveBuffer);
+            _webSocket?.Dispose();
+            _disposed = true;
```

Comment:
Rather than adding another piece of state to track, you could also null out the buffer, and then only return it when it's not null.